### PR TITLE
Add support for Word file attachment.

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Our file upload tooltip had an hardcoded list of extensions that was both too limited and wrong.
Added support for uploading word document.
Refactored so there is only one declaration of supported mime-type & file extensions.
Use this almost everywhere possible.

Noticed that we use a different list and extractions methods in `MultipleDocumentsUpload`, we may consider unifying all this because right now we support a different set of documents types depending on where you are in the product.

Remove deprecated fake mime-type: `file_attachment` & `slack_thread_content` (everything was backfilled)

## Risk

Break file upload.

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/lib/api/files/preprocessing.ts [hunk 1]
```diff
@@ -335,6 +427,10 @@ const processingPerContentType: PreprocessingPerContentType = {
     conversation: storeRawText,
     avatar: notSupportedError,
   },
+  "text/x-markdown": {
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply front/lib/api/files/preprocessing.ts:430]: What you are saying is that we rely on the file extension and that's it ?
I added text/x-markdown as it seems that text/markdown && text/x-markdown co-exists.

Fine by me to remove x-mardown.

[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/api/files/preprocessing.ts:430]: > What you are saying is that we rely on the file extension and that's it ?
🙈 

Just wanted to make sure that we are aware of the difference, if any between those two? Did you get the mime type while testing? How did you find out?

File: types/src/front/content_fragment.ts [hunk 2]
```diff
@@ -10,38 +18,8 @@ export type ContentFragmentContextType = {
...
-// after backfilling data
-const supportedTextContentFragmentType = [
-  "text/plain",
-  "text/csv",
-  "text/tsv",
-  "text/comma-separated-values",
-  "text/tab-separated-values",
-  "text/markdown",
-  "application/pdf",
-  "file_attachment",
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply types/src/front/content_fragment.ts:23]: I'm not sure to understand the nuance(s).
I can add "file_attachment" to the `supportedContentFragmentType` a little bit below if that's appropriate.

File: types/src/front/files.ts [hunk 3]
```diff
@@ -44,27 +44,54 @@ export function ensureFileSize(
...
 
 // Supported content types for images.
-const supportedImageContentTypes = ["image/jpeg", "image/png"] as const;
+const supportedImage = {
+  "image/jpeg": [".jpg", ".jpeg"],
+  "image/png": [".png"],
+} as const;
 
-const supportedUploadableContentType = [
+const uniq = <T>(arr: T[]): T[] => Array.from(new Set(arr));
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply types/src/front/files.ts:69]: doesn't seems to be available in dust-types.

[the remark this answers is not in the document]
Comment (flvndvd) [reply types/src/front/files.ts:68]: You can definitely add it 😉

File: front/lib/api/files/preprocessing.ts [hunk 4]
```diff
@@ -126,18 +134,26 @@ async function createPdfTextStream(buffer: Buffer) {
 
   const pages = extractionRes.value;
 
+  if (pages.length === 0) {
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply front/lib/api/files/preprocessing.ts:137]: I can't remember, removing.

File: front/lib/api/files/preprocessing.ts [hunk 5]
```diff
@@ -311,8 +330,16 @@ type PreprocessingPerContentType = {
 };
 
 const processingPerContentType: PreprocessingPerContentType = {
+  "application/msword": {
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply front/lib/api/files/preprocessing.ts:333]: Yes, upload .doc with the mime type and was able to interact about the file.

File: types/src/front/files.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Fraggle) [reply types/src/front/files.ts:68]: For one function, I'd rather keep it lean ;-)

**Review by flvndvd — APPROVED** [review]

**Pull request merged**